q2_mlab/_preprocess.py

```diff
@@ -0,0 +1,98 @@
+import pandas as pd
+import biom
+from qiime2 import Metadata
+
+
+def print_datasize(table, metadata):
+    biom_table = table.view(biom.Table)
+    dataframe = metadata.to_dataframe()
+    print("\nTable Shape: " + str(biom_table.shape))
+    print("Metadata Shape: " + str(dataframe.shape) + "\n")
+
+
+def clean_metadata(df: pd.DataFrame, target_variable, discrete):
+    # metadata categories that are used in this pipeline are assumed to be
+    # cleaned upfront. For classification, the only allowed values are 0 and 1.
+    # For regression, the allowed values are any real numbers
+    df.dropna(axis=0, subset=[target_variable], inplace=True)
+    # Enforce numeric for continuous variables
+    if not discrete:
+        df[target_variable] = pd.to_numeric(df[target_variable],
+                                            errors='coerce')
+    subset_df = df.loc[:, [target_variable]]
```

Comment:
Dropping NAs should be done at the end.

```suggestion
    # Enforce numeric for continuous variables
    if not discrete:
        df[target_variable] = pd.to_numeric(df[target_variable],
                                            errors='coerce')
    subset_df = df.loc[:, [target_variable]]
    df.dropna(axis=0, subset=[target_variable], inplace=True)
```